packages/source-iottwinmaker/src/property-value/provider.ts

```
@@ -0,0 +1,76 @@
+import { DataRequest, Provider, ProviderObserver, DataBase, DataStreamBase, Primitive } from '@iot-app-kit/core';
+import { QueryClient, QueryObserver } from '@tanstack/react-query';
+
+import { IoTTwinMakerClient } from '@aws-sdk/client-iottwinmaker';
+import { getPropertyValueByEntity } from './client/getPropertyValueByEntity';
+import { TwinMakerStaticDataQuery } from './types';
+
+/**
+ * Provider for TwinMaker time series data
+ */
+export class TwinMakerPropertyValueDataProvider implements Provider<DataBase[]> {
+  public tmClient: IoTTwinMakerClient;
+  public queryClient: QueryClient;
+
+  public observer: QueryObserver<DataBase[], unknown, DataBase[], DataBase[], TwinMakerStaticDataQuery[]>;
+
+  private _unsubscribes: (() => void)[] = [];
+
+  constructor(
+    queryClient: QueryClient,
+    tmClient: IoTTwinMakerClient,
+    {
+      request,
+      queries,
+    }: {
+      request: DataRequest;
+      queries: TwinMakerStaticDataQuery[];
+    }
+  ) {
+    this.queryClient = queryClient;
+    this.tmClient = tmClient;
+    this.observer = new QueryObserver(queryClient, {
+      queryKey: queries,
+      queryFn: async () => {
+        try {
+          const promises: Promise<DataStreamBase<Primitive>[]>[] = [];
+          queries.forEach((query) => {
+            promises.push(getPropertyValueByEntity({ client: tmClient, query: query }));
+          });
+          return [{ dataStreams: (await Promise.all(promises)).flat() }];
+        } catch (_err) {
+          // TODO: return error
+          return [];
```

Maybe return an error object instead of returning the empty array?

---

error handling is my next task